### PR TITLE
[BugFix] use --chown=starrocks:starrocks when COPY in Dockerfile

### PR DESCRIPTION
**Pull request by yandongxiao**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docker/dockerfiles/be/be-ubi.Dockerfile [hunk 1]
```diff
@@ -33,22 +33,24 @@ RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certifica
...
-# Run as starrocks user
-ARG USER=starrocks
-ARG GROUP=starrocks
-RUN groupadd --gid 1000 $GROUP && useradd --no-create-home --uid 1000 --gid 1000 \
-             --shell /usr/sbin/nologin $USER  \
-    && chown -R $USER:$GROUP /opt/starrocks
-USER $USER
-ENV USER=$USER
\ No newline at end of file
+RUN mkdir -p $STARROCKS_ROOT/be/storage && ln -sfT be $STARROCKS_ROOT/cn
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply docker/dockerfiles/be/be-ubi.Dockerfile:56]: `chown -R $USER:$GROUP $STARROCKS_ROOT` has been executed before.

File: docker/dockerfiles/fe/fe-ubi.Dockerfile [hunk 2]
```diff
@@ -33,22 +33,24 @@ RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certifica
...
-# Run as starrocks user
-ARG USER=starrocks
-ARG GROUP=starrocks
-RUN groupadd --gid 1000 $GROUP && useradd --no-create-home --uid 1000 --gid 1000 \
-             --shell /usr/sbin/nologin $USER  \
-    && chown -R $USER:$GROUP /opt/starrocks
-USER $USER
-ENV USER=$USER
\ No newline at end of file
+RUN mkdir -p /opt/starrocks/fe/meta
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply docker/dockerfiles/fe/fe-ubi.Dockerfile:56]: `chown -R $USER:$GROUP $STARROCKS_ROOT` has been executed before.

File: docker/dockerfiles/be/be-ubi.Dockerfile [hunk 3]
```diff
@@ -33,22 +33,24 @@ RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certifica
...
 ENV JAVA_HOME=/usr/lib/jvm/java-openjdk
 
+RUN touch /.dockerenv
+
 WORKDIR $STARROCKS_ROOT
 
+# Run as starrocks user
+ARG USER=starrocks
+ARG GROUP=starrocks
+RUN groupadd --gid 1000 $GROUP && useradd --home-dir /nonexistent --uid 1000 --gid 1000 \
```
Comment (kevincai) [inline docker/dockerfiles/be/be-ubi.Dockerfile:43]: I think we can --home-dir to --no-create-home

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by kevincai** [comment]
https://cookbook.openshift.org/users-and-role-based-access-control/why-do-my-applications-run-as-a-random-user-id.html#:~:text=Within%20the%20output%20of%20oc%20describe%20for%20the,ID%20in%20the%20range%20assigned%20to%20the%20project.

File: docker/dockerfiles/be/be-ubi.Dockerfile [hunk 4]
```diff
@@ -33,22 +33,24 @@ RUN yum install -y java-1.8.0-openjdk-devel tzdata openssl curl vim ca-certifica
...
 ENV JAVA_HOME=/usr/lib/jvm/java-openjdk
 
+RUN touch /.dockerenv
+
 WORKDIR $STARROCKS_ROOT
 
+# Run as starrocks user
+ARG USER=starrocks
+ARG GROUP=starrocks
+RUN groupadd --gid 1000610000 $GROUP && useradd --no-create-home --uid 1000610000 --gid 1000610000 \
```
Comment (kevincai) [inline docker/dockerfiles/be/be-ubi.Dockerfile:43]: keep uid/gid 1000 for now. need some additional fix on start/stop scripts and fe/be for openshift adaption.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by kevincai — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**